sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -31,6 +46,21 @@ object MetalsPlugin extends AutoPlugin {
           "-P:semanticdb:failures:warning",
           s"-P:semanticdb:sourceroot:${(ThisBuild / baseDirectory).value}"
         )
+    },
+    javaSemanticdbEnabled := bspEnabled.value,
+    javacOptions ++= {
+      if (javaSemanticdbEnabled.value)
+        javaSemanticdbOptions.value
+      else
+        Nil
+    },
+    allDependencies ++= {
```

Comment:
This should go to `libraryDependencies` even if it is not a library. In sbt all the compiler plugins are added to `libraryDependencies` (see [here](https://github.com/sbt/sbt/blob/f12a199a1aa58496ade139adcf8a7a8eba94b968/main/src/main/scala/sbt/Defaults.scala#L4323-L4324))

```suggestion
    libraryDependencies ++= {
```

Reply:
Looks like it works fine with allDependencies.
I took it from [`SemanticdbPlugin`](https://github.com/sbt/sbt/blob/f12a199a1aa58496ade139adcf8a7a8eba94b968/main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala#L37)

Reply:
Well maybe it's done on purpose to "hide" the dependency. So it's fine.